src/main/java/pingis/controllers/LiveChallengeController.java

```diff
@@ -186,5 +194,88 @@ public RedirectView closeChallenge(@PathVariable Long challengeId,
     return new RedirectView("/user");
   }
 
+  @RequestMapping("/newArcadeChallenge")
+  public String newArcadeChallenge(Model model) {
+    return "newarcadechallenge";
+  }
+
+  @RequestMapping(value = "/createArcadeChallenge", method = RequestMethod.POST)
+  public RedirectView createArcadeChallenge(RedirectAttributes redirectAttributes,
+      String challengeName,
+      String challengeDesc,
+      String realm,
+      int level) {
+
+    Challenge arcadeChallenge = new Challenge(
+        challengeName,
+        userService.getCurrentUser(),
+        challengeDesc,
+        ChallengeType.ARCADE);
+    logger.info("Created new arcade challenge: " + arcadeChallenge.toString());
+    if (realm == "BEGINNER") {
+      arcadeChallenge.setRealm(Realm.BEGINNER);
+      logger.info("Realm: BEGINNER");
+    } else if (realm == "OBJECTORIENTED") {
+      arcadeChallenge.setRealm(Realm.OBJECTORIENTED);
+      logger.info("Realm: OBJECTORIENTED");
+    } else if (realm == "DATASTRUCTURES") {
+      arcadeChallenge.setRealm(Realm.DATASTRUCTURES);
+      logger.info("Realm: DATASTRUCTURES");
+    } else {
+      logger.info("Realm " + realm + " does not exist. Redirecting to /error.");
+      return new RedirectView("/error");
+    }
```

Comment:
Minor nitpick: use `logger.info("Realm {} does not exist.", realm);` instead of `logger.info("Realm " + realm + " does not exist.");`. The former way is easier to read, write, and search for. (It's also more performant but that doesn't matter much in our case.)